main.tf

```diff
@@ -32,50 +29,52 @@ resource "random_id" "suffix" {
   byte_length = 4
 }
 
-
-module "gke_auth" {
-  source  = "terraform-google-modules/kubernetes-engine/google//modules/auth"
-  version = "~> 10.0"
-
-  project_id   = module.project_services.project_id
-  cluster_name = module.gke.name
-  location     = module.gke.location
-
-  depends_on = [time_sleep.sleep_for_cluster_fix_helm_6361]
-}
+# Retrieve an access token
+data "google_client_config" "provider" {}
 
 provider "helm" {
   kubernetes {
-    load_config_file       = false
-    cluster_ca_certificate = module.gke_auth.cluster_ca_certificate
-    host                   = module.gke_auth.host
-    token                  = module.gke_auth.token
+    cluster_ca_certificate = base64decode(google_container_cluster.gitlab.master_auth[0].cluster_ca_certificate)
+    host                   = "https://${google_container_cluster.gitlab.endpoint}"
+    token                  = data.google_client_config.provider.access_token
   }
 }
 
 provider "kubernetes" {
-  load_config_file = false
-
-  cluster_ca_certificate = module.gke_auth.cluster_ca_certificate
-  host                   = module.gke_auth.host
-  token                  = module.gke_auth.token
+  cluster_ca_certificate = base64decode(google_container_cluster.gitlab.master_auth[0].cluster_ca_certificate)
+  host                   = "https://${google_container_cluster.gitlab.endpoint}"
+  token                  = data.google_client_config.provider.access_token
 }
 
 // Services
-module "project_services" {
```

Comment:
Was there an issue with using the `project_services` submodule? You can use the latest version for compat with 1.0.

Reply:
`google_project_service` is part of the official Google provider, and in my opinion has a couple of advantages:
* enables fine-grained dependencies
* makes maintenance easier by avoid to manage module versions

Reply:
@compalmanel we maintain all modules in `terraform-google-modules`, so we should continue to use upstream modules when possible.

Reply:
Also we should try to use implicit dependencies where possible as opposed to explicit
https://github.com/terraform-google-modules/terraform-google-gke-gitlab/blob/c56a13c3933de1f38504686963a9e3ccbf5bd5b0/main.tf#L278-L282

Reply:
> @compalmanel we maintain all modules in terraform-google-modules, so we should continue to use upstream modules when possible.

The modules on `terraform-google-modules` don't support Terraform 1.0 and the official Terraform Google provider has all of the required functionality.

> Also we should try to use implicit dependencies where possible as opposed to explicit

Agree 100%, but I had to add these dependencies to make everything work correctly.

Reply:
> terraform-google-modules don't support Terraform 1.0

We are in the process but most popular modules (project-factory, gke) should already be 1.0 compatible. Did you run into an issue using project-services with 1.0?

Reply:
The big question here is, why is it necessary to use modules if the official provider supports the required functionality?

Reply:
All modules are more or less packaging of resources from an official provider. The value add is a cleaner abstraction and devex as I don't have to repeat config (`disable_dependent_services = true` etc). We also promote using modules to showcase how individual modules can be composed to form higher order modules like this.